soda/core/soda/sodacl/sodacl_parser.py

```diff
@@ -1184,99 +1199,59 @@ def __parse_table_filter_section(self, antlr_table_filter_header, header_str, he
-        if isinstance(header_content, dict):
-            automated_monitoring_cfg = AutomatedMonitoringCfg(self.data_source_name, self.location)
-            data_source_scan_cfg = self.sodacl_cfg._get_or_create_data_source_scan_cfgs(self.data_source_name)
-            data_source_scan_cfg.add_monitoring_cfg(automated_monitoring_cfg)
-            automated_monitoring_cfg.data_source_name = header_content.get("data_source")
-            tables = header_content.get("tables")
-            if isinstance(tables, list):
-                for table in tables:
-                    if table.startswith("exclude "):
-                        exclude_table_expression = table[len("exclude ") :]
-                        automated_monitoring_cfg.exclude_tables.append(exclude_table_expression)
+    def __parse_tables(self, header_content, cfg):
+        cfg.data_source_name = header_content.get("data_source")
+        tables = header_content.get("tables")
+        if isinstance(tables, list):
+            for table in tables:
+                if table.startswith("exclude "):
+                    exclude_table_expression = table[len("exclude ") :]
+                    cfg.exclude_tables.append(exclude_table_expression)
+                else:
+                    if table.startswith("include "):
+                        include_table_expression = table[len("include ") :]
                     else:
-                        if table.startswith("include "):
-                            include_table_expression = table[len("include ") :]
-                        else:
-                            include_table_expression = table
-                        automated_monitoring_cfg.include_tables.append(include_table_expression)
-            else:
-                self.logs.error(
-                    'Content of "tables" must be a list of include and/or exclude expressions', location=self.location
-                )
-
-            find_anomalies = header_content.get("find anomalies")
-            if find_anomalies:
-                row_count = find_anomalies.get("row count")
-                if isinstance(row_count, bool) and row_count == False:
-                    automated_monitoring_cfg.row_count = False
-                schema = find_anomalies.get("schema")
-                if isinstance(schema, bool) and schema == False:
-                    automated_monitoring_cfg.schema = False
+                        include_table_expression = table
+                    cfg.include_tables.append(include_table_expression)
         else:
             self.logs.error(
-                f'Skipping section "{header_str}" because content is not an object/dict',
-                location=self.location,
+                'Content of "tables" must be a list of include and/or exclude expressions', location=self.location
             )
 
+    @assert_header_content_is_dict
+    def __parse_automated_monitoring_section(self, header_str, header_content):
```

Comment:
`header_str` is unused - Same in other other places. `__parse_discover_tables_section`, `__parse_profile_columns_section`

Reply:
yes but it's used in `assert_header_content_is_dict`. Do you have a recommendation how I can remove it?

Reply:
Ah - you mean it is used in decorator - sorry